roles/matrix-base/tasks/setup_matrix_base.yml

```diff
@@ -19,6 +19,59 @@
     mode: '0660'
   when: "matrix_vars_yml_snapshotting_enabled | bool"
 
+- name: Save current git-repo status on the target to aid with restoring in case of problems
+  block:
+    - name: Get local git hash 
+      delegate_to: 127.0.0.1
+      become: false
+      register: git_describe
+      shell:
+        git describe
+            --always
+            --tags
+            --dirty
+            --long
+            --all
+
+    - set_fact: 
+        git_hash: "{{ git_describe.stdout }}"
+
+    - name: Git hash
+      debug:
+        msg: "Git hash: {{ git_hash }}"
+    
+    - name: Save git hash in git_hash.yml
+      become: false
+      local_action: 
+        copy
+        content="git_hash_last_run{{ ":" }} {{ git_hash }}\n"
+        dest="{{ matrix_vars_yml_snapshotting_src }}/git_hash.yml"
```

Comment:
`matrix_vars_yml_snapshotting_src` defaults to:

```yaml
matrix_vars_yml_snapshotting_src: "{{ inventory_dir }}/host_vars/{{ inventory_hostname }}/vars.yml"
```

So.. using a path like `{{ inventory_dir }}/host_vars/{{ inventory_hostname }}/vars.yml/git_hash.yml` doesn't seem to make much sense.

Also.. There's probably no need to save the git hash locally - you can just save it remotely.

Reply:
understood, i saw your comment on the other PR as well. I guess somewhere during my mods i modified the original `matrix_vars_yml_snapshotting_src` (removed the vars.yml) as it nicely give me the root where we need to store the git hash.
Any suggestion on how to get the correct snapshot directory without having to cut the `vars.yml` off of `matrix_vars_yml_snapshotting_src`? Or are you ok with changing `matrix_vars_yml_snapshotting_src` to only have the dir and than accept my other PR as well? that seems to be the only location where `matrix_vars_yml_snapshotting_src` is used currently. I prefer the latter.

I'll change it to store the `git_hash.yml` only on the target.

Reply:
ok, scratch that directory comment.... we don't need the directory as we won't save it locally.
I need some coffee